modAionImpl/src/org/aion/zero/impl/AionBlockchainImpl.java

```diff
@@ -1148,7 +1149,14 @@ private RetValidPreBlock generatePreBlock(IAionBlock block) {
                         block.getNrgLimit(),
                         LOGGER_VM,
                         getPostExecutionWorkForGeneratePreBlock());
-        List<AionTxExecSummary> executionSummaries = executor.execute();
+
+        List<AionTxExecSummary> executionSummaries = null;
+        try {
+            executionSummaries = executor.execute();
+        } catch (VMException e) {
+            LOG.error(e.toString());
```

Comment:
I think this will only print the exception, without the full stack trace. A meaningful message should be added and the stack trace, for example: `LOG.error("Shutdown due to a VM fatal error.",e);`

Reply:
updated